File: src/scanthosegates/campaign/listeners/SalvagingListener.java
package scanthosegates.campaign.listeners;

import com.fs.starfarer.api.Global;
import com.fs.starfarer.api.campaign.SectorEntityToken;
import com.fs.starfarer.api.campaign.listeners.DiscoverEntityListener;
import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import scanthosegates.campaign.econ.abilities.CryosleeperScanner;
import scanthosegates.campaign.econ.abilities.HypershuntScanner;

public class SalvagingListener implements DiscoverEntityListener {
    private static final Logger log = Global.getLogger(SalvagingListener.class);
    static {log.setLevel(Level.ALL);}

    @Override
    public void reportEntityDiscovered(SectorEntityToken entity){
        CryosleeperScanner.tryCreateCryosleeperReportCustom(entity, log, true, true);
        HypershuntScanner.tryCreateHypershuntReport(entity, log, true, true);
    }
}

File: src/org/aero/scanThoseGates/campaign/listeners/SalvagingListener.kt
package org.aero.scanThoseGates.campaign.listeners

import com.fs.starfarer.api.Global
import com.fs.starfarer.api.campaign.SectorEntityToken
import com.fs.starfarer.api.campaign.listeners.DiscoverEntityListener
import org.aero.scanThoseGates.campaign.abilities.CryosleeperScanner.Companion.tryCreateCryosleeperReportCustom
import org.aero.scanThoseGates.campaign.abilities.HypershuntScanner.Companion.tryCreateHypershuntReport
import org.apache.log4j.Level

class SalvagingListener : DiscoverEntityListener {
    override fun reportEntityDiscovered(entity: SectorEntityToken) {
        tryCreateCryosleeperReportCustom(entity, log, true, true)
        tryCreateHypershuntReport(entity, log, true, true)
    }

    companion object {
        private val log = Global.getLogger(SalvagingListener::class.java)

        init {
            log.level = Level.ALL
        }
    }
}
